cpp/include/cudf/join/filtered_join.hpp

```diff
@@ -0,0 +1,153 @@
+
+  /**
+   * @copydoc filtered_join(cudf::table_view const&, null_equality, bool, rmm::cuda_stream_view)
+   *
+   * @param load_factor The desired ratio of filled slots to total slots in the hash table, must be
+   * in range (0,1]. For example, 0.5 indicates a target of 50% occupancy. Note that the actual
+   * occupancy achieved may be slightly lower than the specified value.
+   */
+  filtered_join(cudf::table_view const& build,
+                null_equality compare_nulls  = null_equality::EQUAL,
+                bool reuse_left_table        = false,
+                double load_factor           = 0.5,
+                rmm::cuda_stream_view stream = cudf::get_default_stream());
+
+  /**
+   * @brief Returns a vector of row indices corresponding to a semi-join
+   * between the specified tables.
+   *
+   * The returned vector contains the row indices from the left table
+   * for which there is a matching row in the right table. Note that the left table
+   * is the build table if `reuse_left_table` is set to true, and is the probe table
+   * otherwise.
+   *
+   * @code{.pseudo}
+   * TableA: {{0, 1, 2}}
+   * TableB: {{1, 2, 3}}
+   * Result: {1, 2}
+   * @endcode
+   *
+   * @param probe The probe table
+   * @param stream CUDA stream used for device memory operations and kernel launches
+   * @param mr Device memory resource used to allocate the returned table and columns' device memory
+   *
+   * @return A vector `left_indices` that can be used to construct
+   * the result of performing a left semi join
+   */
+  [[nodiscard]] std::unique_ptr<rmm::device_uvector<size_type>> semi_join(
+    cudf::table_view const& probe,
+    rmm::cuda_stream_view stream      = cudf::get_default_stream(),
+    rmm::device_async_resource_ref mr = cudf::get_current_device_resource_ref()) const;
+
+  /**
+   * @brief Returns a vector of row indices corresponding to a anti-join
+   * between the specified tables.
+   *
+   * The returned vector contains the row indices from the left table
+   * for which there are no matching rows in the right table. Note that the left table
+   * is the build table if `reuse_left_table` is set to true, and is the probe table
+   * otherwise.
+   *
+   * @code{.pseudo}
+   * TableA: {{0, 1, 2}}
+   * TableB: {{1, 2, 3}}
+   * Result: {1, 2}
+   * @endcode
+   *
+   * @param probe The probe table
+   * @param stream CUDA stream used for device memory operations and kernel launches
+   * @param mr Device memory resource used to allocate the returned table and columns' device memory
+   *
+   * @return A vector `left_indices` that can be used to construct
+   * the result of performing a left anti join
+   */
+  [[nodiscard]] std::unique_ptr<rmm::device_uvector<size_type>> anti_join(
+    cudf::table_view const& probe,
+    rmm::cuda_stream_view stream      = cudf::get_default_stream(),
+    rmm::device_async_resource_ref mr = cudf::get_current_device_resource_ref()) const;
+
+ private:
+  bool const _reuse_left_table;
```

Comment:
```suggestion
  bool _reuse_left_table;
```
[Data members: Never const](https://quuxplusone.github.io/blog/2022/01/23/dont-const-all-the-things/#data-members-never-const)

Reply:
Doesn't matter, since this class can't be copied nor moved.